### DISCUSSION TITLE
Calc whitespace best practice

### DISCUSSION THREAD

**Discussion by m-murphyy** [tailwindlabs/tailwindcss#12147] [Help]
We are trying to decide on the best practice for how to handle whitespace in the `calc()` function within custom styles. 

Where we're coming from:
1. The calc function (outside of tailwind) needs whitespace around operators per the mdn docs (see the first bullet point under the "Notes" section [here](https://developer.mozilla.org/en-US/docs/Web/CSS/calc#notes)).
2. The tailwind docs say that "when an arbitrary value needs to contain a space, use an underscore (_) instead and Tailwind will automatically convert it to a space at build-time" ([source](https://tailwindcss.com/docs/adding-custom-styles#handling-whitespace)). Note that `calc()` usage is not provided as one of the code examples in that section.
3. We have noticed that you do not _have_ to use underscores in invocations of calc within arbitrary values to get the expected result. You can also just remove the whitespace entirely and it has worked in all of our use cases.
4. I was unable to find an answer on the docs or in previous discussions on github.

What we want to know:
As best practice, should we use underscores in our `calc()` calls when using arbitrary values? Have we just gotten lucky that our current uses have not needed underscores to work properly? Or is calc a special case for tailwind? Is there somewhere in the docs explaining this that we missed?

Thank you!

**Comment by wongjn** [comment]
Tailwind internally *does* try to resolve white space in `calc()` for you and I believe you should be able to rely on this behavior. I have reported several issues in the past with the whitespace not being added in some instances and the maintainers did dutifully fix those cases.
